lib/std/special/compiler_rt/clzsi2.zig

```diff
@@ -0,0 +1,150 @@
+// Ported from:
+//
+// https://github.com/llvm-mirror/compiler-rt/blob/f0745e8476f069296a7c71accedd061dce4cdf79/lib/builtins/clzsi2.c
+// https://github.com/llvm-mirror/compiler-rt/blob/f0745e8476f069296a7c71accedd061dce4cdf79/lib/builtins/arm/clzsi2.S
+const builtin = @import("builtin");
+
+// Precondition: a != 0
+fn __clzsi2_generic(a: i32) callconv(.C) i32 {
+    @setRuntimeSafety(builtin.is_test);
+
+    var x: c_uint = @bitCast(c_uint, a);
+    var t: c_int = @intCast(c_int, @boolToInt((x & 0xFFFF0000) == 0)) << 4; // if (x is small) t = 16 else 0
```

Comment:
Minor nit, can you please put the comments on their own line if it becomes too long?

Reply:
I can, but this is how it was in the original C code

Reply:
But this is not the original C code :)